From 647d6ac3efd57a2c6cab8611b0684b60f40b063d Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Mon, 22 Dec 2014 15:41:16 +0800
Subject: [PATCH 185/196] atomisp2: Fix a memory leak problem

It will lead memory leak, when system free dis_buf and s3a_buf.

Change-Id: I46b1f30fd9b736b4e036c72a4a936553772cef9e
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Tracked-on: https://jira01.devtools.intel.com/browse/IMINAN-14504
Reviewed-on: https://icggerrit.ir.intel.com/33758
Reviewed-by: Shuguang Gong <shuguang.gong@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c       | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 925d9de..6f74874 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -1003,7 +1003,8 @@ int atomisp_alloc_css_stat_bufs(struct atomisp_sub_device *asd,
 	int count;
 	unsigned int i;
 
-	if (list_empty(&asd->s3a_stats)) {
+	if (list_empty(&asd->s3a_stats) &&
+		asd->params.curr_grid_info.s3a_grid.enable) {
 		count = ATOMISP_CSS_Q_DEPTH +
 		        ATOMISP_S3A_BUF_QUEUE_DEPTH_FOR_HAL;
 		dev_dbg(isp->dev, "allocating %d 3a buffers\n", count);
@@ -1024,7 +1025,8 @@ int atomisp_alloc_css_stat_bufs(struct atomisp_sub_device *asd,
 		}
 	}
 
-	if (list_empty(&asd->dis_stats)) {
+	if (list_empty(&asd->dis_stats) &&
+		asd->params.curr_grid_info.dvs_grid.enable) {
 		count = ATOMISP_CSS_Q_DEPTH + 1;
 		dev_dbg(isp->dev, "allocating %d dis buffers\n", count);
 		while (count--) {
-- 
1.9.1

